app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ActiveRecord::Base
+  has_many :comments, foreign_key: :author_id
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
```

Comment:
- [ ] In this `has_many` association, you are saying that a user has many comments, posts, and likes through a foreign key `author_id` but in your migrations what you have as the foreign key is `user_id`. This is causing errors while creating posts, comments, and likes. Kindly fix this by referencing the correct column `user_id` that you have in your migrations or you can generate a new migration and update the reference keys of comments, posts, and likes table to `author_id`.

Reply:
This has been resolved.